common/environment/environment.cpp

```diff
@@ -1698,16 +1697,18 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
 #endif
                     if (targetIp.ipequals(serverIP))
                     {
-                        // OK the target is a valid machine on the white list
+                        // OK the target is a valid machine in the server list we have a right drop zone candidate
                         isServerFound = true;
                         break;
                     }
                 }
                 if (isServerFound)
-                    break;
-
-                // Path matched, but machine didn't
-                dropZone = nullptr;
+                    // Keep this candidate drop zone if its directory path is shorter than we already have
+                    if (dropzonePathLen > fullDropZoneDir.length())
+                    {
+                       dropzonePathLen = fullDropZoneDir.length();
+                       dropZone = candidateDropZone;
```

Comment:
Is there any reason to have this condition code after the ForEach loop above and not do it at the time (instead of) setting isServerFound ?
i.e.
```
    if (targetIp.ipequals(serverIP))
    {
        // Keep this candidate drop zone if its directory path is shorter than we already have
        if (dropzonePathLen > fullDropZoneDir.length())
        {
            dropzonePathLen = fullDropZoneDir.length();
            dropZone = candidateDropZone;
        }
        break;
    }
```